src/Playwright.Tests/BrowserTypeConnectTests.cs

```diff
@@ -80,71 +82,396 @@ public void TearDown()
+            var page = await browser.NewPageAsync();
+            await browser.CloseAsync();
+
+            Assert.AreEqual(browser.IsConnected, false);
+            var exception = await PlaywrightAssert.ThrowsAsync<PlaywrightException>(async () => await page.GotoAsync(Server.EmptyPage));
+            StringAssert.Contains("Connection closed (Browser closed)", exception.Message);
         }
 
         [PlaywrightTest("browsertype-connect.spec.ts", "should reject waitForSelector when browser closes")]
-        [Ignore("SKIP WIRE")]
-        public void ShouldRejectWaitForSelectorWhenBrowserCloses()
+        public async Task ShouldRejectWaitForSelectorWhenBrowserCloses()
         {
+            var browser = await BrowserType.ConnectAsync(_browserServer.WSEndpoint);
+            var context = await browser.NewContextAsync();
+            var page = await context.NewPageAsync();
+            var watchdog = page.WaitForSelectorAsync("div");
+            await browser.CloseAsync();
+
+            Assert.AreEqual(browser.IsConnected, false);
+            var exception = await PlaywrightAssert.ThrowsAsync<PlaywrightException>(async () => await watchdog);
+            Assert.That(exception.Message, Contains.Substring("Target closed"));
         }
 
         [PlaywrightTest("browsertype-connect.spec.ts", "should emit close events on pages and contexts")]
-        [Ignore("SKIP WIRE")]
-        public void ShouldEmitCloseEventsOnPagesAndContexts()
+        public async Task ShouldEmitCloseEventsOnPagesAndContexts()
         {
+            var browser = await BrowserType.ConnectAsync(_browserServer.WSEndpoint);
+            var context = await browser.NewContextAsync();
+            var page = await context.NewPageAsync();
+            bool pageClosed = false, contextClosed = false;
+            page.Close += (_, e) => pageClosed = true;
+            context.Close += (_, e) => contextClosed = true;
+
+            await browser.CloseAsync();
+            Assert.AreEqual(pageClosed, true);
+            Assert.AreEqual(contextClosed, true);
+            Assert.AreEqual(browser.IsConnected, false);
         }
 
         [PlaywrightTest("browsertype-connect.spec.ts", "should terminate network waiters")]
-        [Ignore("SKIP WIRE")]
-        public void ShouldTerminateNetworkWaiters()
+        public async Task ShouldTerminateNetworkWaiters()
+        {
+            var browser = await BrowserType.ConnectAsync(_browserServer.WSEndpoint);
+            var context = await browser.NewContextAsync();
+            var page = await context.NewPageAsync();
+            var task = TaskUtils.WhenAll(page.WaitForRequestAsync(Server.EmptyPage), page.WaitForResponseAsync(Server.EmptyPage), browser.CloseAsync());
+            try
+            {
+                await task;
+            }
+            catch (Exception)
+            {
+                foreach (var exception in task.Exception.InnerExceptions)
```

Comment:
You test will pass (false positive) if there are no exceptions.

Reply:
Have fixed this test